### ISSUE TITLE
cmdlet queue blocked and flies locked by failed cmdlets

### ISSUE THREAD

**Issue by qiyuangong** [Intel-bigdata/SSM#638]
cmdlet queue will be blocked when there are `dead` cmdlet. There `dead` cmdlets cannot be finished correctly or be marked as `DONE`, their states are always `PENDING`. Worse more, these cmdlets will locked related files and caused `Failed to submit cmdlet for file:`.

```
Failed to submit cmdlet for file: /test123/test7
```

```
mysql> select * from cmdlets where state=1;          
+-----+-----+------+-------+------------+---------------+--------------------+
| cid | rid | aids | state | parameters | generate_time | state_changed_time |
+-----+-----+------+-------+------------+---------------+--------------------+
|   5 |   1 | 5    |     1 | archive    | 1499664649051 |      1499664649051 |
|   9 |   2 | 9    |     1 | allssd     | 1499665359660 |      1499665359660 |
|  34 |   1 | 34   |     1 | archive    | 1499670406608 |      1499670406608 |
|  38 |   2 | 38   |     1 | allssd     | 1499670521633 |      1499670521633 |
+-----+-----+------+-------+------------+---------------+--------------------+
4 rows in set (0.00 sec)

```

**Comment by huafengw** [comment]
Checked the log, somehow the Cmdlet status update message is lost or missing

**Comment by huafengw** [comment]
Fixed in #659
